src/NServiceBus.Core/Persistence/Learning/SagaPersister/SagaStorageFile.cs

```diff
@@ -12,10 +14,36 @@ class SagaStorageFile : IDisposable
         {
             this.fileStream = fileStream;
             this.manifest = manifest;
+            textWriter = new StreamWriter(fileStream, Encoding.Unicode);
+            jsonWriter = new JsonTextWriter(textWriter)
+            {
+                CloseOutput = true
+            };
+            textReader = new StreamReader(fileStream, Encoding.Unicode);
+            jsonReader = new JsonTextReader(textReader)
+            {
+                CloseInput = true
+            };
 
             lastModificationSeenAt = File.GetLastWriteTimeUtc(fileStream.Name);
         }
 
+        public void Dispose()
+        {
```

Comment:
This could be simplified to 
```
jsonWriter.Close();
jsonReader.Close();
```
Because `Close` on the json classes calls `Close` on the `Stream` classes, and calling `Close` on those calls `Dispose` on them, which also disposes the underlying streams.